### DISCUSSION TITLE
SSL Error connecting to planetscale -  [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:992)

### DISCUSSION THREAD

**Discussion by BeigeSponge** [planetscale/discussion#604] [Bug Reports]
When connecting via my EC2 instance I am getting the following error:

` There was an error connecting to the database: 2055: Lost connection to MySQL server at 'server', system error: 1 [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:992)`
 
 I have updated openssh to version 9.1.1 and have been unable to replicate this error locally. Does anyone have any ideas why I'm running into this? I am using mysql-connector.
 
My cert path is:` /etc/pki/tls/certs/ca-bundle.crt`. Also, for reference my connection code looks like: 

```
    try:
        connection = mysql.connector.connect(
            host=db_host,
            user=db_username,
            password=db_password,
            database=db_name,
            autocommit=True,
            ssl_ca=db_cert_path,
        )
    except mysql.connector.Error as e:
        logger.error(f"There was an error connecting to the database: {e}")
        raise
    return connection
```

**Comment by orware** [comment]
Hello @BeigeSponge,

Did you mean OpenSSH or OpenSSL in your comment above? 

I would expect the OpenSSL version to be more related to the possibility of the older `TLSV1` protocol being used, particularly if your EC2 instance is possibly running a slightly older distribution.

If you are using the latest version of that particular Python connector:
https://pypi.org/project/mysql-connector-python/

And OpenSSL has been updated that should hopefully default to using TLS 1.2+ and allow the connections to succeed.
